Src/FluentAssertions/Equivalency/EnumerableEquivalencyValidator.cs

```diff
@@ -62,35 +64,73 @@ private void AssertElementGraphEquivalency<T>(object[] subjects, T[] expectation
             unmatchedSubjectIndexes = new List<int>(subjects.Length);
             unmatchedSubjectIndexes.AddRange(Enumerable.Range(0, subjects.Length));
 
+            if (OrderingRules.IsOrderingStrictFor(context))
+            {
+                AssertElementGraphEquivalencyWithStrictOrdering(subjects, expectations);
+            }
+            else
+            {
+                AssertElementGraphEquivalencyWithLooseOrdering(subjects, expectations);
+            }
+        }
+
+        private void AssertElementGraphEquivalencyWithStrictOrdering<T>(object[] subjects, T[] expectations)
+        {
+            int failedCount = 0;
             foreach (int index in Enumerable.Range(0, expectations.Length))
             {
                 T expectation = expectations[index];
 
-                if (!OrderingRules.IsOrderingStrictFor(context))
+                using (context.TraceBlock(path =>
+                    $"Strictly comparing expectation {expectation} at {path} to item with index {index} in {subjects}"))
                 {
-                    using (context.TraceBlock(path => $"Finding the best match of {expectation} within all items in {subjects} at {path}[{index}]"))
+                    bool succeed = StrictlyMatchAgainst(subjects, expectation, index);
+                    if (!succeed)
                     {
-                        LooselyMatchAgainst(subjects, expectation, index);
+                        failedCount++;
+                        if (failedCount >= FailedItemsFastFailThreshold)
+                        {
+                            context.TraceSingle(path =>
+                                $"Fail failing strict order comparison of collection after {FailedItemsFastFailThreshold} items failed at {path}");
```

Comment:
🤔 I _think_ you mean that you already found 10 mismatches and decided to abort the comparison for performance reasons, right?

Reply:
"Fail failing strict order comparison" - jeez what was I thinking? Fixed.